ledger/ledger-api-common/src/test/suite/scala/com/digitalasset/ledger/api/validation/EventQueryServiceRequestValidatorTest.scala

```diff
@@ -0,0 +1,119 @@
+// Copyright (c) 2023 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.ledger.api.validation
+
+import com.daml.error.{ContextualizedErrorLogger, NoLogging}
+import com.daml.ledger.api.messages.event
+import com.daml.ledger.api.v1.value.Value
+import com.daml.ledger.api.v1.{event_query_service, value => api}
+import com.daml.lf.value.{Value => Lf}
+import io.grpc.Status.Code._
+import org.mockito.MockitoSugar
+import org.scalatest.wordspec.AnyWordSpec
+
+class EventQueryServiceRequestValidatorTest
+    extends AnyWordSpec
+    with ValidatorTestUtils
+    with MockitoSugar {
+
+  private implicit val noLogging: ContextualizedErrorLogger = NoLogging
+
+  private val validator = new EventQueryServiceRequestValidator(PartyNameChecker.AllowAllParties)
+
+  "EventQueryServiceRequestValidator" when {
+
+    "validating event by contract id requests" should {
+
+      val expected = event.GetEventsByContractIdRequest(
+        contractId = contractId,
+        requestingParties = Set(party),
+      )
+
+      val req = event_query_service.GetEventsByContractIdRequest(
+        contractId.coid,
+        expected.requestingParties.toSeq,
+      )
+
+      "pass on valid input" in {
+        validator.validateEventsByContractId(req) shouldBe Right(expected)
```

Comment:
I forgot we have multiple patterns for validating the requests. I must admit that this "applicative" approach seems the most elegant to me.